volpe/__main__.py

```diff
@@ -13,43 +13,83 @@
 """
 
 import os
-from sys import version_info
+from sys import version_info, platform
 from docopt import docopt
 
 
 def install():
-    # Look up user path in registry.
-    import winreg
-
-    reg_key = winreg.OpenKey(winreg.HKEY_CURRENT_USER, r"Environment")
-    path = winreg.QueryValueEx(reg_key, "Path")[0]
-
     path_to_volpe = os.path.abspath(os.path.dirname(__file__))
     path_to_volpe_root = os.path.dirname(path_to_volpe)
+    path_to_volpe_bin = os.path.join(path_to_volpe_root, "bin")
 
-    # Update the .bat file
-    print("Updating batch file with volpe directory.")
-    with open(os.path.join(path_to_volpe_root, "volpe.bat"), "w") as OPATH:
-        OPATH.writelines(["@echo off\n", f"python {path_to_volpe} %*"])
-    print("=====")
+    batch_file_path = os.path.join(path_to_volpe_bin, "volpe.bat")
+    bash_file_path = os.path.join(path_to_volpe_bin, "volpe")
 
-    if path_to_volpe_root in path:
-        print("Volpe is already on PATH.")
+    if not os.path.isdir(path_to_volpe_bin):
+        os.makedirs(path_to_volpe_bin)
 
-        if path_to_volpe_root not in os.environ["PATH"]:
-            print("=====")
-            print("Please restart this console for changes to take effect.")
+    if platform == "win32" and not os.path.isfile(batch_file_path):
+        # Update the batch file.
+        print("Creating batch file with volpe directory.")
+        with open(batch_file_path, "w") as batch_file:
+            batch_file.writelines(["@echo off\n", f"python {path_to_volpe} %*"])
 
-    else:
-        # Add Volpe path to user path.
-        print(f"Adding {path_to_volpe_root} to user PATH.")
-        path_including_volpe = path + os.pathsep + path_to_volpe_root
-        print("Setting local user PATH.")
-        os.system(f'SETX Path "{path_including_volpe}"')
+    elif platform == "linux" and not os.path.isfile(bash_file_path):
+        # Update the bash file.
+        print("Creating bash file with volpe directory.")
+        with open(bash_file_path, "w") as bash_file:
+            bash_file.writelines(["#!/bin/bash\n", f"python {path_to_volpe} $@"])
+
+        # Add executable permissions.
+        print("Adding executable permission to bash file with chmod")
+        import stat
+        st = os.stat(bash_file_path)
+        os.chmod(bash_file_path, st.st_mode | stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH)
+
+    if path_to_volpe_bin in os.environ["PATH"]:
+        print("Volpe is already on PATH.")
+        return
+
+    if platform == "win32":
+        # Make sure it is not on PATH before trying to add it
+        import winreg
+        reg_key = winreg.OpenKey(winreg.HKEY_CURRENT_USER, "Environment")
+        path = winreg.QueryValueEx(reg_key, "Path")[0]
+
+        if path_to_volpe_bin not in path:
```

Comment:
Why is this checked twice?

Reply:
Because it is possible that is in the path environment variable, but not in `os.environ["PATH"]` e.g. because the console hasn't been restarted.

Reply:
So the first check is not really necessary then?

Reply:
true, but it skips having to look stuff up in the registry

Reply:
Shrug